src/GithubMergeTool/GithubMergeTool.cs

```diff
@@ -38,32 +39,49 @@ public class GithubMergeTool
         /// Create a merge PR.
         /// </summary>
         /// <returns>
-        /// null if the merge PR was completed without error. Otherwise,
-        /// the response which had an error is returned. Note that a response
-        /// with an <see cref="HttpStatusCode" /> value 422 is returned if a PR
-        /// cannot be created because the <paramref name="srcBranch"/> creates
-        /// all the commits in the <paramref name="destBranch"/>.
+        /// (true, null) if the PR was created without error.
+        /// (false, null) if the PR wasn't created due to a PR already existing
+        /// or if the <paramref name="destBranch"/> contains all the commits
+        /// from <paramref name="srcBranch"/>.
+        /// (false, error response) if there was an error creating the PR.
         /// </returns>
-        public async Task<HttpResponseMessage> CreateMergePr(
+        public async Task<(bool prCreated, HttpResponseMessage error)> CreateMergePr(
             string repoOwner,
             string repoName,
             string srcBranch,
             string destBranch)
         {
+            string prTitle = $"Merge {srcBranch} to {destBranch}";
+
+            // Check to see if there's already a PR
+            // TODO: handle pagination
+            // This handles the 100 most recent PRs, which should work for now
+            HttpResponseMessage prsResponse = await _client.GetAsync(
+                $"repos/{repoOwner}/{repoName}/pulls?state=open&base={destBranch}&per_page=100");
```

Comment:
❗️ This doesn't check if the existing PR is up-to-date.

:memo: Keep in mind that when you merge the most recent PR for a target branch, all other open PRs for that same target branch close automatically. The previous approach created more pull requests, but it allowed for seamless review using either an incremental approach or by simply taking the most recent one.

I would recommend simply not including this feature.

Reply:
@sharwell The creation of multiple PRs created lots of confusion and duplicate work. Furthermore, with this and the auto-merge support we're planning on running this once an _hour_ not once a day, which would make duplicates a non-starter.

Reply:
And some background: right now doing these merges once a day is proving to be a significant bottleneck on code flow. The extra day of latency causes problems with shiproom, ensuring we don't have insertion conflicts between 15.6/15.7, etc.

Reply:
@jasonmalinowski Thanks, that's a sufficient response for me considering I'm not Tiger.